src/Compilers/CSharp/Portable/FlowAnalysis/ReadWriteWalker.cs

```diff
@@ -220,7 +220,28 @@ public override void VisitForEachIterationVariable(BoundForEachStatement node)
 
         public override BoundNode VisitRangeVariable(BoundRangeVariable node)
         {
-            NoteRead(node.RangeVariableSymbol);
+            // Compute the "underlying symbol" for a read of the range variable
+            Symbol underlyingSymbol = GetUnderlyingVariable(node.Value);
+            NoteRead(node.RangeVariableSymbol, underlyingSymbol);
+            return null;
+        }
+
+        private Symbol GetUnderlyingVariable(BoundNode underlying)
+        {
+            while (underlying != null)
+            {
+                switch (underlying.Kind)
+                {
+                    case BoundKind.Parameter:
+                        return ((BoundParameter)underlying).ParameterSymbol;
+                    case BoundKind.PropertyAccess:
+                        underlying = ((BoundPropertyAccess)underlying).ReceiverOpt;
+                        continue;
+                    default:
+                        return null;
```

Comment:
I don't believe we have requirements for its behavior in error scenarios, so I don't think throwing is a good idea.